be/src/exec/vectorized/aggregate/agg_hash_map.h

```diff
@@ -87,6 +87,27 @@ struct AggHashMapWithOneNumberKey {
         }
     }
 
+    template <typename Func>
+    void compute_agg_states_with_limit(size_t chunk_size, const Columns& key_columns, MemPool* pool,
+                                       Func&& allocate_func, Buffer<AggDataPtr>* agg_states, size_t limit,
+                                       std::vector<uint8_t>* not_founds) {
+        DCHECK(!key_columns[0]->is_nullable());
+        auto column = down_cast<ColumnType*>(key_columns[0].get());
+        (*not_founds).assign(chunk_size, 0);
+
+        for (size_t i = 0; i < column->size(); i++) {
```

Comment:
Is this code can be simplified by integrated into `compute_agg_states` function? looks like there are a lot of boilerplate code.

Reply:
it was the same function before, it can be integrated after i make sure that there is no negative performance impact.

Reply:
done